### DISCUSSION TITLE
Can Nixie do this?

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#876] [Q&A] [answered]
I've been running njsPC & njsHA for over a year and am currently building a Nixie to replace a ET 4.  I'm making steady progress on the build and am in the mode of regularly ordering parts (connectors, terminal blocks, etc.) that I didn't realize I needed.  I'm currently awaiting some goodies that will hopefully get me over the finish line.

In the interim, I have a Pi stack on my desk and am configuring as much as I can without it being able to talk to the high voltage relays, RS485 devices, sensors, etc.  I believe I have everything at the Nixie end ready to go--Hat relays seem to be working as expected, temperature sensors are configured (without the probes attached), and feeds appear to be configured correctly.  The Nixie was discovered in Home Assistant & I can operate the relays & get feeds from the HA front end and from dashPanel.  I've setup schedules & just moments ago, the virtual pool filter pump turned on!  Exciting stuff, indeed!!!

My pool has an unusual feature I would like to incorporate into the Nixie, if possible.  We frequently get torrential rainstorms that cause the pool to overflow, due to the combination of heavy downpours and grossly undersized overflow drain lines.

To combat this problem, I incorporated a moisture sensor into the pool filler lid that notifies Home Assistant if the pool is overly full.  Home Assistant then conditionally turns on the filter pump, waits a minute, & then opens a drain valve to lower the water level.  It's a true life saver!

The drain valve is incorporated into the Nixie configuration and appears to work (virtually at this point).  I have a 10 minute egg timer to prevent emptying the pool completely.  If possible, I would like to be able to simply turn on the drain valve switch and have it turn on the pump if it's not already running, pause to get past priming, open the valve, time out, & close the valve.

What I  don't know how to do with the Nixie is ensure the filter pump is running and turn it on as needed prior to opening the valve & starting the egg timer.  It seems to be similar to, say, turning master cleaner (timeout to speed up the pump), or ramping up the pump for a gas heater.  These functions are, of course, built into Nixie so folks like me can just take advantage of the functionality without additional configuration on my part.

Can you kindly let me know if Nixie can do what I need for the drain valve & either tell me how to do it or point me to a solution?

Many thanks for your help!

**Comment by tagyoureit** [comment]
There are a couple examples of custom rule processors that can show you how to accomplish this:
- https://github.com/tagyoureit/nodejs-poolController/discussions/687
- https://github.com/tagyoureit/nodejs-poolController/discussions/527

You should always use extreme caution when automating any drain functionality.  An empty pool is a sad pool.  There can always be some edge cases around this... like what if you start a timer but it overlaps with a schedule start/stop time.  So make sure you think through all contingencies.

**Comment by FlaMike** [comment]
As always, thank you so much for your quick & helpful reply!

I understand the problems that may result (are likely to result?) from automatically draining a pool.  Many things can go wrong, indeed!  I am currently using a Node Red flow to activate the drain valve.  It has a 10 minute timeout that closes the valve.  Of course, if NR goes down or the communication path between the boxes fails, it could be a bad day.  As a backup to hopefully cover that scenario, I use a 10 minute egg timer for the valve on the ET.

I was thinking that putting the logic to control the valve on the Nixie may reduce/eliminate a failure mode by eliminating the communication link from NR to the Nixie.  And the first thing I did when configuring the valve (again, on my desk) was to add the 10 minute egg timer.  On my desk, it appears to be working as intended, but...

All that being said, the pool "technician" who replaced the ET motherboard that failed a few months ago blew it big time!  He got confused when configuring the new board.  My wife & I walked our dogs the next morning.  When we returned home, perhaps 45 minutes later, the water level in our pool was down over 12".  He confused some of the circuits & opened the drain when the filter pump started.  There was no egg timer set, so had we not returned home & had I not gone to the backyard where I heard the water exiting the drain hose, our pool would have been bone dry and/or the pump may have overheated & failed within a couple of hours.  To say the least, he ain't been back since & I am working on my Nixie build so I never have to call a "technician" again!

If you know anything else I can do to fail safe automatically draining the pool after 10 minutes or so, I am all ears.

Thanks again for all of your help & creating & supporting  njsPC and all its associated goodies!  I cannot tell you how much I appreciate it!

**Reply by rstrouse** [reply, marked as answer]
Actually, Nixie uses a latching mechanism that will unlatch the relay if comms are lost.  As well if you use the normally open side of the relay as the default/closed position, this will cause the valve to be closed if power is lost.  So you can do the same with a feature on Nixie where it has a 10 minute egg timer that will forcibly close after 10 minutes or if there is failure.

**Reply by FlaMike** [reply]
Thanks--good to know!  I will certainly be sure to use the NO side of the relay for the default position of the valve!  Because I have a one body pool, the drain valve is the only actuated valve in the system--not even a spa here, just a lap pool.  The egg timer is already on the gestating Nixie and it appears to work as expected--on the desktop, of course.

In the short run (at the very least until Nixie has been installed & gets  "settled in") I'll continue to use the NR flow to initiate the drain valve & filter pump as I am doing currently.  Using the NO side of the relay should provide additional safety for the drain valve should communication from NR/HA go  down.  I have a couple of UPSes in place for the HA/NR box and networking equipment, primarily to keep things alive until the whole house generator comes up to speed.  I *do* start to worry about having too many layers of "protection" from disaster in place, since they can complicate the overall system.  Still, they appear to be independent enough as to not step on each other's toes.

**Reply by FlaMike** [reply]
Noodling a bit more on your very helpful and  comforting response, I have a naive follow-up question that comes to mind as I'm finally starting to wire up the Nixie box on the workbench.  When you suggest using the NO side of the relay default/closed position--is that wiring the Nixie as shown in the wiki?  Or, do I need to swap the NO/NC connections?  I just want to get this wired correctly out of the gate so there aren't any "surprises" when Nixie goes live.  

Thank you both @rstrouse  and @tagyoureit  so much for your help.  I hope you both have a fabulous 2024!!!

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/e321be01-7e2f-4bb8-9f3b-8394651fd2ab)

**Reply by rstrouse** [reply]
As it is in the wiki.

**Reply by FlaMike** [reply]
Excellent!  Thank you for confirming the wiring for the valve!
